src/lib/krb5/krb/sendauth.c

```diff
@@ -127,26 +127,27 @@ krb5_sendauth(krb5_context context, krb5_auth_context *auth_context,
     }
 
     if (ap_req_options & AP_OPTS_USE_SUBKEY) {
-        /* Provide some more fodder for random number code.
-           This isn't strong cryptographically; the point here is
-           not to guarantee randomness, but to make it less likely
-           that multiple sessions could pick the same subkey.  */
-        char rnd_data[1024];
+        /* Provide some more fodder for random number code.  This isn't strong
+         * cryptographically; the point here is not to guarantee randomness,
+         * but to make it less likely that multiple sessions could pick the
+         * same subkey.
+         */
+        struct sockaddr rnd_data;
```

Comment:
``struct sockaddr`` won't necessarily be large enough to hold the result of getpeername() or getsockname().  rnd_data should be of type ``struct sockaddr_storage`` and then its address should be casted with ss2sa() when passing it to getpeername/getsockname.